moveit_ros/planning/moveit_cpp/src/moveit_cpp.cpp

```diff
@@ -276,6 +276,25 @@ MoveItCpp::execute(const std::string& group_name, const robot_trajectory::RobotT
   return moveit_controller_manager::ExecutionStatus::RUNNING;
 }
 
+bool MoveItCpp::terminatePlanningPipeline(std::string const& pipeline_name)
+{
+  try
+  {
+    auto const& planning_pipeline = planning_pipelines_.at(pipeline_name);
+    if (planning_pipeline->isActive())
+    {
+      planning_pipeline->terminate();
```

Comment:
`terminate()` returns true, even if the planner is not running. https://github.com/ros-planning/moveit2/blob/main/moveit_core/planning_interface/include/moveit/planning_interface/planning_interface.h#L128
We should check for `false`, though.

Reply:
Good point, is it enough to just check the activity of the planner instance itself or do we need to make the whole pipeline more interrupt-able, for example how long can `planning_scene->isPathValid(...)`? The isActive() function checks on the pipeline level while terminate() only checks on the planner level.